editor/app/ems/[incident_number]/page.tsx

```diff
@@ -0,0 +1,205 @@
+    return Array.from(new Set(relatedCrashPks));
+  }, [ems_pcrs]);
+
+  /**
+   * Get all people records linked to crashes associated with these incidents
+   */
+  const { data: matchingPeople } = useQuery<PeopleListRow>({
+    query: relatedCrashPks ? GET_MATCHING_PEOPLE : null,
+    variables: {
+      crash_pks: relatedCrashPks,
+    },
+    typename: "people_list_view",
+  });
+
+  const onSaveCallback = useCallback(async () => {
+    await refetch();
+  }, [refetch]);
+
+  /**
+   * Memoize the additional components props for related record tables
+   */
+  const linkRecordButtonProps: EMSLinkRecordButtonProps = useMemo(
+    () => ({
+      onClick: (emsPcr) => {
+        setSelectedEmsPcr((prevEmsPcr) => {
+          return prevEmsPcr?.id === emsPcr?.id ? null : emsPcr;
+        });
+      },
+      selectedEmsPcr: selectedEmsPcr,
+    }),
+    [selectedEmsPcr]
+  );
+
+  const linkToPersonButtonProps: EMSLinkToPersonButtonProps = useMemo(
+    () => ({
+      onClick: (emsId, personId, crashPk) => {
+        updateEMSIncident({
+          id: emsId,
+          person_id: personId,
+          crash_pk: crashPk,
+        })
+          .then(() => refetch())
+          .then(() => {
+            setSelectedEmsPcr(null);
+          });
+      },
+      selectedEmsPcr: selectedEmsPcr,
+    }),
+    [updateEMSIncident, selectedEmsPcr, refetch]
+  );
+
+  if (error) {
+    console.error(error);
+  }
+
+  /**
+   * Use the first EMS record as the "incident"
+   */
+  const incident = ems_pcrs?.[0];
+
+  /**
+   * Set the title of the page inside the HTML head element
+   */
+  useEffect(() => {
+    if (incident) {
+      document.title = `EMS ${incident.incident_number} - ${incident.incident_location_address}`;
+    }
+  }, [incident]);
+
+  if (!ems_pcrs) {
+    return;
+  }
+
+  if (!incident) {
+    notFound();
+  }
+
+  return (
+    <>
+      <Row>
+        <Col className="d-flex fs-3 align-items-center mb-3">
+          <FaTruckMedical className="me-2" />
+          <span>{incident.incident_location_address}</span>
+        </Col>
+      </Row>
+      <Row>
+        <Col sm={12} md={6} lg={4} className="mb-3">
+          <DataCard<EMSPatientCareRecord>
+            record={incident}
+            isValidating={isValidating}
+            title="Summary"
+            columns={emsDataCards.summary}
+            mutation={""}
+            onSaveCallback={onSaveCallback}
+          />
+        </Col>
+        <Col sm={12} className="mb-3">
+          <RelatedRecordTable<EMSPatientCareRecord, EMSLinkRecordButtonProps>
+            records={ems_pcrs}
+            isValidating={isValidating}
+            noRowsMessage="No crashes found"
+            header="EMS patient(s)"
+            columns={emsDataCards.patient}
+            mutation={UPDATE_EMS_INCIDENT}
+            onSaveCallback={onSaveCallback}
+            rowActionComponent={EMSLinkRecordButton}
+            rowActionComponentAdditionalProps={linkRecordButtonProps}
```

Comment:
This PR extends the `rowActionComponent` so that it can accept additional props, which enables the table to share state with other components. Getting the typing right was pretty tricky, and I have some doubts about how well this scales. I'll add my thoughts on this in the `RelatedRecordTable` component